python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -754,9 +754,18 @@ def _utilization_score(node_resources: ResourceDict,
 
     util_by_resources = []
     for k, v in node_resources.items():
+        # Don't divide by zero.
+        if v < 1:
+            # Could test v == 0 on the nose, but v < 1 feels safer.
+            # (Note that node resources are integers.)
+            continue
         util = (v - remaining[k]) / v
         util_by_resources.append(v * (util**3))
 
+    # Could happen if node_resources has only zero values.
+    if not util_by_resources:
+        return None
+
```

Comment:
can we change the type of the returned type?
instead of `float` to use `Optional[float]` or something